### ISSUE TITLE
My Cars dialog: Any change to Car Name field should immediately update Active Car

### ISSUE THREAD

**Issue by EVGrokker** [aesculus/EVTO-App-Feedback#439]
If I open the My Cars dialog and change the car model (S, X, 3), the Car Name field immediately updates to a default value, e.g. 'My Model X'. When that happens, the Active Car value in the picker should change immediately to the value in the Car Name field.

Additionally, when the user edits the Car Name field manually, upon exiting the field, the Active Car value in the picker should be updated immediately to the value in the Car Name field.

Bottom line: any changes to the Car Name field, whether manual or automatic, should immediately update the current selection in the Active Car picker.

**Comment by aesculus** [comment]
I think we are mixing some items. The Car Name field is just a label. Don't get confused by the fact that I default the label to My Model X or My Model S when you select a new car model. I just did that so it did not say My Tesla, and have 5 of those you could not tell apart.

So if you change the text in the Car Name field, it has no implications to any other aspect of the My Cars data.

**Comment by EVGrokker** [comment]
I agree that it doesn't affect anything internally, but it does affect the user's perception of the program's behavior. The next time they come back into the dialog, that's the text that should be in the picker, so why not update it immediately? Also, it's the text that's topmost in the Main menu, so updating it immediately confirms for the user that things will be correct when they exit.

**Comment by aesculus** [comment]
Are you experiencing a lag when the panel opens that the car name or model takes time to be right?

**Comment by EVGrokker** [comment]
No perceptible problem that I've noticed.

**Comment by aesculus** [comment]
OK. So I am confused by this statement: 

>The next time they come back into the dialog, that's the text that should be in the picker, so why not update it immediately?

If you last left it as a Model X, was the picker not a Model X when you returned?

**Comment by EVGrokker** [comment]
Sorry for the confusion. There are three fields involved in this discussion:
- The drop-down picker, which lets you pick which of the 5 possible car slots you're actively using;
- The name text field, which is your nickname for the car
- The radio button selection of the model (which my mockup proposes to change to a picker, to accommodate the Y and semi (haha) ).

My comment was meant to convey that when editing the text field to change the nickname, the drop-down picker should be updated immediately to reflect the change.

And if the user selects a different model, the immediate effect would be that the nickname would change to the default name for that model (which it currently does), AND the picker name for the active slot would be changed to match.

**Comment by aesculus** [comment]
> My comment was meant to convey that when editing the text field to change the nickname, the drop-down picker should be updated immediately to reflect the change.

I may be getting closer to understanding your issue but not 100%. Are you saying you want me to monitor the Car Name field and when you type one in that matches an existing one of the 5 cars to switch the active car?

**Comment by aesculus** [comment]
BTW I can see the car model control heading to a select picker soon too (once Model Y and or Semi come out that will be the limit). We can only accommodate one more car model in the tablet display before we run out of room.

**Comment by EVGrokker** [comment]
Uhm, no.

Let's suppose that I've previously visited the My Cars dialog and configured a car in slot 1. It's a Model X, I've nicknamed it Thunder. 

When I now open the dialog, the picker displays 'Thunder' in slot 1 as the active car.

If I want to start configuring car 2, I would select 'My Car 2' in the picker and proceed to customize it. As soon as I nickname it, the picker name would be updated to match.

The only mechanism to select the active car is selecting it from the picker.

There are two ways a car's name switches from the default 'My Car [x]' in the picker list, either assigning a car other than an S, or changing the nickname.

Hope that makes sense.

**Comment by aesculus** [comment]
> If I want to start configuring car 2, I would select 'My Car 2' in the picker and proceed to customize it. As soon as I nickname it, the picker name would be updated to match.

OK. I was clueless until you gave me the detailed scenario.  I will try to implement this.

**Comment by aesculus** [comment]
Can it be done after you leave the field rather than character by character?

**Comment by EVGrokker** [comment]
Yes, that's what I meant but didn't convey - upon the name field losing focus, update the picker text.

**Comment by aesculus** [comment]
OK. Probably easy to do that.

I just noticed another potential issue (you tell me). If you select car 2 for example but don't save and hit cancel, the active car still changes to car 2.  Do you see that as an issue?

**Comment by EVGrokker** [comment]
I think it's an acceptable risk. 'My Car 2' would be listed in the Main menu header. If they file a bug report, you'd see that the data was all default. Maybe it would be a good idea to include the nickname and slot number (e.g. 1/5) in the feedback report to be able to catch that condition.

**Comment by aesculus** [comment]
Implemented in V1.2 (37)
